src/units/animation.hpp

```diff
@@ -29,14 +30,19 @@ class unit_animation
 		/** Shouldn't be used so only declared. */
 		unit_animation();
 	public:
-		typedef enum { MATCH_FAIL=-10 , DEFAULT_ANIM=-9} variation_type;
-		typedef enum { HIT, MISS, KILL, INVALID} hit_type;
+		enum variation_type { MATCH_FAIL=-10 , DEFAULT_ANIM=-9};
+		MAKE_ENUM(hit_type,
+			(HIT, "hit")
+			(MISS, "miss")
+			(KILL, "kill")
+			(INVALID, "invalid")
+		);
 
 		static const std::vector<std::string>& all_tag_names();
 		static void fill_initial_animations( std::vector<unit_animation> & animations, const config & cfg);
 		static void add_anims( std::vector<unit_animation> & animations, const config & cfg);
 
-		int matches(const display &disp,const map_location& loc,const map_location& second_loc,const unit* my_unit,const std::string & event="",const int value=0,hit_type hit=INVALID,const attack_type* attack=nullptr,const attack_type* second_attack = nullptr, int value2 =0) const;
+	int matches(const display &disp,const map_location& loc,const map_location& second_loc,const unit* my_unit,const std::string & event="",const int value=0,hit_type hit=hit_type::INVALID,const attack_type* attack=nullptr,const attack_type* second_attack = nullptr, int value2 =0) const;
```

Comment:
Ugh, that would be because I have XCode set to re-indent when I type :, and this file is indented weirdly.